pkg/golinters/nolintlint/nolintlint.go

```diff
@@ -2,31 +2,47 @@ package nolintlint
 
 import (
 	"fmt"
-	"go/ast"
 	"sync"
 
 	"golang.org/x/tools/go/analysis"
 
+	"github.com/golangci/golangci-lint/pkg/golinters/internal"
+
 	"github.com/golangci/golangci-lint/pkg/config"
 	"github.com/golangci/golangci-lint/pkg/goanalysis"
-	"github.com/golangci/golangci-lint/pkg/golinters/nolintlint/internal"
+	nolintlint "github.com/golangci/golangci-lint/pkg/golinters/nolintlint/internal"
 	"github.com/golangci/golangci-lint/pkg/lint/linter"
-	"github.com/golangci/golangci-lint/pkg/result"
 )
 
-const LinterName = "nolintlint"
+const LinterName = nolintlint.LinterName
 
 func New(settings *config.NoLintLintSettings) *goanalysis.Linter {
 	var mu sync.Mutex
 	var resIssues []goanalysis.Issue
 
+	var needs nolintlint.Needs
+	if settings.RequireExplanation {
+		needs |= nolintlint.NeedsExplanation
+	}
+	if settings.RequireSpecific {
+		needs |= nolintlint.NeedsSpecific
+	}
+	if !settings.AllowUnused {
+		needs |= nolintlint.NeedsUnused
+	}
+
+	lnt, err := nolintlint.NewLinter(needs, settings.AllowNoExplanation)
+	if err != nil {
+		internal.LinterLogger.Fatalf("asasalint: create analyzer: %v", err)
+	}
+
 	analyzer := &analysis.Analyzer{
-		Name: LinterName,
+		Name: nolintlint.LinterName,
```

Comment:
If we use the constant to set this value, shouldn't we refer to it as well? If not I think we can drop the constant and always use `nolintlint.LinterName`

Reply:
we cannot do that because it's an internal package.